LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/Regions/RegionCN470.cs

```diff
@@ -171,15 +171,17 @@ public override bool TryGetDownstreamChannelFrequency(Rxpk upstreamChannel, out
         /// <summary>
         /// Logic to get the correct downstream transmission frequency for region CN470.
         /// <param name="upstreamFrequency">The frequency at which the message was transmitted.</param>
-        /// <param name="dataRate">The upstream data rate.</param>
+        /// <param name="upstreamDataRate">The upstream data rate.</param>
         /// <param name="deviceJoinInfo">Join info for the device, if applicable.</param>
         /// </summary>
-        public override bool TryGetDownstreamChannelFrequency(double upstreamFrequency, ushort dataRate, out double downstreamFrequency, DeviceJoinInfo deviceJoinInfo)
+        public override bool TryGetDownstreamChannelFrequency(double upstreamFrequency, out double downstreamFrequency, ushort? upstreamDataRate = null, DeviceJoinInfo deviceJoinInfo = default)
```

Comment:
How come the upstream datarate could be nullable, I don't see it being used in this method

Reply:
It is used only in the US region. The downstream frequency in US depends on the data rate. In other regions it's not used. Do you have another suggestion how to do this better?

Reply:
forgot about this ;)